cardano-cli/test/cardano-cli-golden/Test/Golden/Help.hs

```diff
@@ -0,0 +1,98 @@
+{-# LANGUAGE OverloadedStrings #-}
+
+{- HLINT ignore "Redundant id" -}
+
+module Test.Golden.Help
+  ( helpTests
+  ) where
+
+import           Prelude hiding (lines)
+
+import           Control.Monad (forM_, unless)
+import           Data.Text (Text)
+import           Hedgehog (Property)
+import           Hedgehog.Extras.Stock.OS (isWin32)
+import           System.FilePath ((</>))
+import           Test.Cardano.CLI.Util (execCardanoCLI, propertyOnce)
+import           Text.Regex (Regex, mkRegex, subRegex)
+
+import qualified Data.Char as Char
+import qualified Data.List as List
+import qualified Data.Text as Text
+import qualified Hedgehog as H
+import qualified Hedgehog.Extras.Test.Base as H
+import qualified Hedgehog.Extras.Test.Golden as H
+import qualified Test.Cardano.CLI.Util as H
+
+ansiRegex :: Regex
+ansiRegex = mkRegex "\\[[0-9]+m"
+
+filterAnsi :: String -> String
+filterAnsi line = subRegex ansiRegex stripped ""
+  where stripped = filter (/= '\ESC') line
+
+{- HLINT ignore "Use camelCase" -}
+
+extractCmd :: Text -> [Text]
+extractCmd = id
+  . takeWhile nonSwitch
+  . Text.split Char.isSpace
+  . Text.strip
+  where nonSwitch :: Text -> Bool
+        nonSwitch s =
+          case Text.unpack (Text.take 1 s) of
+            (c:_) -> Char.isAlpha c
+            [] -> False
+
+-- | Test that converting a @cardano-address@ Byron signing key yields the
+-- expected result.
+golden_HelpAll :: Property
+golden_HelpAll =
+  propertyOnce . H.moduleWorkspace "help" $ \_ -> do
+    -- These tests are not run on Windows because the cardano-cli usage
+    -- output is slightly different on Windows.  For example it uses
+    -- "cardano-cli.exe" instead of "cardano-cli".
+    unless isWin32 $ do
+      helpFp <- H.note "test/cardano-cli-golden/files/golden/help.cli"
+
+      help <- filterAnsi <$> execCardanoCLI
+        [ "help"
+        ]
+
+      H.diffVsGoldenFile help helpFp
+
+third :: (a, b, c) -> c
+third (_, _, c) = c
+
+golden_HelpCmds :: Property
+golden_HelpCmds =
+  propertyOnce . H.moduleWorkspace "help-commands" $ \_ -> do
+    -- These tests are not run on Windows because the cardano-cli usage
+    -- output is slightly different on Windows.  For example it uses
+    -- "cardano-cli.exe" instead of "cardano-cli".
+    unless isWin32 $ do
+      help <- filterAnsi <$> execCardanoCLI
+        [ "help"
+        ]
+
+      let lines = Text.lines $ Text.pack help
+      let usages = List.filter (not . null) $ extractCmd . Text.drop 19 <$> List.filter (Text.isPrefixOf "Usage: cardano-cli ") lines
```

Comment:
Text.drop 19 removes "Usage: cardano-cli "? Adding a comment saying this would be useful.

Reply:
I've introduced a new function which returns the magic number `19`.